backend/ethereum/subscription/eventsub.go

```diff
@@ -199,25 +205,25 @@ func (s *EventSub) readFuture(ctx context.Context, sink chan<- *Event) error {
 			case <-ctx.Done():
 				return ctx.Err()
 			case sink <- event:
-			case <-s.closed:
+			case <-s.closer.Closed():
 				return nil
 			}
 		case err := <-s.watchSub.Err():
 			err = cherrors.CheckIsChainNotReachableError(err)
 			return err
 		case <-ctx.Done():
 			return ctx.Err()
-		case <-s.closed:
+		case <-s.closer.Closed():
 			return nil
 		}
 	}
 }
 
 // Close closes the sub and frees associated resources.
-// Should be called exactly once and panics otherwise.
-// Must not be called if the construction function returned with an error.
+// Can be called more than once, but should be called at least once.
+// Is thread safe.
 func (s *EventSub) Close() {
-	close(s.closed)
-	s.watchSub.Unsubscribe()
-	s.filterSub.Unsubscribe()
+	if err := s.closer.Close(); !pkgsync.IsAlreadyClosedError(err) {
+		log.WithError(err).Error("could not close EventSub")
```

Comment:
Regarding the error ignore here:  
I changed it to return the error but the change would propagate up to the usage of the `RegisteredSub` so i would rather do this in a separate issue.

Reply:
Move to #162